src/main/java/core/basesyntax/MainApp.java

```diff
@@ -1,5 +1,42 @@
 package core.basesyntax;
 
+abstract class Machine {
+    public abstract void doWork();
+    public abstract void stopWork();
+}
+class Truck extends Machine {
+    public void doWork() {
+        System.out.println("Truck start");
+    }
+    public void stopWork() {
+        System.out.println("Truck stop");
+    }
+}
+class Bulldozer extends Machine {
+    public void doWork() {
+        System.out.println("Bulldozer start");
+    }
+
+    public void stopWork() {
+        System.out.println("Bulldozer stop");
+    }
+}
+class Excavator extends Machine {
+    public void doWork() {
+        System.out.println("Excavator start");
+    }
+    public void stopWork() {
+        System.out.println("Excavator stop");
+    }
+}
 public class MainApp {
+    public static void main(String[] args) {
+        Machine[] spisoc = {new Truck(), new Bulldozer(), new Excavator()};
```

Comment:
Use abstract references instead of specific ones where possible. Instead of creating a new array with the type 'Machine[]', you should use 'List<Machine>' to take advantage of polymorphism and make the code more flexible.